src/conda/devcontainer-feature.json

```diff
@@ -22,6 +22,7 @@
     },
     "containerEnv": {
         "CONDA_DIR": "/opt/conda",
-        "PATH": "${PATH}:${CONDA_DIR}/bin:"
+        "CONDA_SCRIPT":"${CONDA_DIR}/etc/profile.d/conda.sh",
+        "PATH": "${PATH}:${CONDA_DIR}/bin"
```

Comment:
I think this pattern that we've adopted here (using one containerEnv inside another) -_might_ be dangerous moving forward, since I think that the order this object is emitted into the intermediate Dockerfile is important. Right now the tooling ends up emitting in the provided order

<img width="951" alt="image" src="https://user-images.githubusercontent.com/23246594/190026942-a8a360d8-9df3-473f-a92a-79dc4c6cfd87.png">


<img width="907" alt="image" src="https://user-images.githubusercontent.com/23246594/190026886-220ecb18-ee0d-488c-8ab2-db688f431423.png">


<img width="872" alt="image" src="https://user-images.githubusercontent.com/23246594/190026847-1515b7ee-e847-42c9-aa74-067365926c87.png">

--- 

Notably, this output is different when using docker exec directly
<img width="796" alt="image" src="https://user-images.githubusercontent.com/23246594/190027174-b1d3897e-68a2-411e-968d-86e70587d777.png">

Reply:
Changing the order to:

```
    "containerEnv": {
        "PATH": "${PATH}:${CONDA_DIR}/bin",
        "CONDA_SCRIPT":"${CONDA_DIR}/etc/profile.d/conda.sh",
        "CONDA_DIR": "/opt/conda"
    }
```

The auto-generated test fails:
<img width="466" alt="image" src="https://user-images.githubusercontent.com/23246594/190027569-414bd439-93ca-41c1-b203-24f89356736e.png">

<img width="943" alt="image" src="https://user-images.githubusercontent.com/23246594/190027661-32a014dd-d0e6-4f99-a7e2-3dbad8a2ce2b.png">

<img width="954" alt="image" src="https://user-images.githubusercontent.com/23246594/190027723-8e5654eb-282a-43d6-8cb6-130d26cd1952.png">

<img width="885" alt="image" src="https://user-images.githubusercontent.com/23246594/190027777-9cb90ccb-e401-40f7-b5ae-dbb0e2a27933.png">

<img width="767" alt="image" src="https://user-images.githubusercontent.com/23246594/190027796-e16c621d-1e10-47f0-8cb9-05e6fd8e61ea.png">

Reply:
It seems to make the most sense to me to encode information about the `$PATH` into the `devcontainer-feature.json`, but we might want to explore a more explicit approach.  Note that this is only showing one feature, but multiple features might alter the `$PATH` variable in different ways.

Reply:
That's a very good point. Changed all the usage of `containerEnv` to no longer depend on another `containerEnv` so that we could avoid the ordering issue.